tests/bluechi_test/machine.py

```diff
@@ -103,15 +103,24 @@ def systemctl_start_and_wait(self, service_name: str, sleep_after: float = 0.0)
 
         return result, output, wait_result
 
-    def install_systemd_service(self, service: Service,
-                                target_dir: str = os.path.join("/", "etc", "systemd", "system")):
-        LOGGER.debug(f"Installing systemd service '{service.name}' to container path '{target_dir}'"
-                     f"with content:\n{service}")
-        self.create_file(target_dir, service.name, str(service))
+    def install_systemd_service(self, service: Service, backup_existing: bool = False, stop_at_cleanup: bool = True):
+        if backup_existing:
+            self._track_changed_file(service.get_directory(), service.get_name())
+
+        LOGGER.debug(f"Installing systemd service '{service.get_name()}' to container path '{service.get_directory()}'"
+                     f"with content:\n{service.to_string()}")
+        self.create_file(service.get_directory(), service.get_name(), service.to_string())
         self.systemctl.daemon_reload()
 
         # keep track of created service file to potentially stop it in later cleanup
-        self.systemctl.tracked_services.append(service.name)
+        self.systemctl.tracked_services.append(service.get_name())
+
+    def load_systemd_service(self, service: Service):
```

Comment:
```suggestion
    def get_systemd_service_from_file(self, service_path: pathlib.Path) -> Service:
```
Passing the Service and updating it feels like C-style. What do you think about passing a `str` or `pathlib.Path` and return a new `Service` instance?

Reply:
In the usage in `restart_with_config_file` it would then be:
```python
service = self.get_systemd_service_from_file(service_path)
```

Reply:
Done